### ISSUE TITLE
AlertNode JSON custom field

### ISSUE THREAD

**Issue by AAbouZaid** [influxdata/kapacitor#1071]
Recently I needed to some data in the JSON payload that AlertNode creates, but I couldn't find a way to do that.
I think we need a way to add more data to AlertNode's JSON payload.

This should be as same as level of main ones like "id", "message", "level" ... etc.
But not just string (maybe list or dict).

```
Each event that gets sent to a handler contains the following alert data:
    ID – the ID of the alert, user defined.
    Message – the alert message, user defined.
    Details – the alert details, user defined HTML content.
    Time – the time the alert occurred.
    Duration – the duration of the alert in nanoseconds.
    Level – one of OK, INFO, WARNING or CRITICAL.
    Data – influxql.Result containing the data that triggered the alert.
```

Info:
* Kapacitor: v1.1

Thanks

**Comment by nathanielc** [comment]
@AAbouZaid What data did you want to add? In other words what was the source of the data?

**Comment by AAbouZaid** [comment]
@nathanielc It could be anything I think.

It could be a fixed string inside TICK script, or something comes form Influx query (just like what we have in "details" where query tags or fields could be used inside it).

**Comment by nathanielc** [comment]
@AAbouZaid @rbetts I believe this can be accomplished via the custom JSON via HTTP post feature added in #1572. 

If this issue is requesting more beyond what can be accomplished with #1572 can we get an example?

**Comment by russorat** [comment]
A trivial usecase: We process a measurement where a certain field is not always defined. We want to supress the error that results in a node failing to process that field if it is undefined. We could start adding default nodes but that can become really convoluted. Just having kapacitor drop the problematic line to the floor *quietly* would be the preferred option.
I've hit situations several times where I've found myself trying doing extra work with a script just to supress the error messages while the script itself is already working properly

**Comment by rbetts** [comment]
Close this; resolved by #1572.
